drivers/i2c/i2c_dw_registers.h

```diff
@@ -13,62 +13,27 @@ extern "C" {
 #endif
 
 /*  IC_CON bits */
-#define IC_CON_TX_INTR_MODE			(1 << 8)
-#define IC_CON_STOP_DET_IFADDR			(1 << 7)
-#define IC_CON_SLAVE_DISABLE			(1 << 6)
-#define IC_CON_RESTART_EN			(1 << 5)
-#define IC_CON_10BIT_ADDR_MASTER		(1 << 4)
-#define IC_CON_10BIT_ADDR_SLAVE			(1 << 3)
-#define IC_CON_SPEED_MASK			(0x3 << 1)
-#define IC_CON_MASTER_MODE			(1 << 0)
-
 union ic_con_register {
-	uint16_t                raw;
+	uint32_t                raw;
 	struct {
-		uint16_t         master_mode : 1 __packed;
-		uint16_t         speed : 2 __packed;
-		uint16_t         addr_slave_10bit : 1 __packed;
-		uint16_t         addr_master_10bit : 1 __packed;
-		uint16_t         restart_en : 1 __packed;
-		uint16_t         slave_disable : 1 __packed;
-		uint16_t         stop_det : 1 __packed;
-		uint16_t         tx_empty_ctl : 1 __packed;
-		uint16_t         rx_fifo_full : 1 __packed;
+		uint32_t         master_mode : 1 __packed;
```

Comment:
Well, here's an underlying problem: It is implementation-defined (C17 J.3.9) in C whether a bit-field type is anything other than a `signed int`, `unsigned int`, or `_Bool` (C17 6.7.2.1).  So this should probably be `unsigned int`.

Hopefully the endianness of the target platforms is consistent with the bit ordering used by the toolchain.

Reply:
With the change of driver style from bitfield struct to read/write access functions, do these remaining fields/structs need to be bitfields/packed at all?